main.rb

```diff
@@ -0,0 +1,146 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each { |n| puts "#{n}!" }
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    arr = to_a
+    while i < arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+  [1, 2, 3].my_each_each_with_index { |_n, x| print x }
```

Comment:
`my_each_each_with_index` doesn't exists.

Reply:
**OPTIONAL** Also, I suggest you remove all the sample you add. They are not really needed.